uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -451,28 +451,41 @@ public static function findOverlappingLeaveRequests($contactID, $fromDate, $toDa
    *   And array of values from the Leave Request Status OptionGroup
    * @param bool $publicHolidaysOnly
    *   When true, will get the balance change only for the Public Holiday Leave Requests
+   * @param bool $expiredOnly
+   *   When true, will only count the balance change for expired days
    *
    * @return array
    */
   public static function getBalanceChangeByAbsenceType(
     $contactID,
     $periodID,
     $leaveRequestStatus = [],
-    $publicHolidaysOnly = false
+    $publicHolidaysOnly = false,
+    $expiredOnly = false
```

Comment:
Since the Brought Forward is part of the entitlement and we can think of accrued TOIL as being part of the entitlement as well, then probably we should have a method on the LeavePeriodEntitlement BAO then that returns the expired balance change. 
It being on the `civicrm_api3_leave_request_getbalancechangebyabsencetype()` makes it look as if the balance change is directly related to Leave Request and its not.
Or probably on an API method on leave_period_entitlement?